uwblueprint/ccbc
# Connect creator profile form to backend

## Notion ticket link

[Creator Profile: Multi-step form](https://www.notion.so/uwblueprintexecs/Creator-Profile-Multi-step-form-ad603a8fc07a43d0a7bac478a22e553c?pvs=4)



## Implementation description
* Overall connect the creator profile to the backend but very huge PR so generally
* When user first saves/submits the form it checks if the creator object exists for this user id - if so it updates the existing creator object otherwise creates a new object and updates
* For every save isReadyForReview is marked as false but for submission it is marked as true and shows up in the admin dashboard for review
* The creator-profile form is continually accessible after submit for updates (not sure when this should be closed)
* A modal is used for submission confirmation + toasts for saves / errors
* Upon load of the form if the user has saved data the data is populated in the form fields
* Also fixed a ton of type issues + had to edit some of the validators that were oudated/prevented in progress creator saves + added a route for getCreatorByUserId 



## Steps to test
1. DELETE all existing creators, the seeding script has both set up with a userId of 1 which isn't a real-life scenario as far as I understand
2. Go through form and edit/save as desired
3. Reloading the form should repopulate all fields and save data
4. Upon submission you can check the admin dashboard for the creator's name to pop up 



## What should reviewers focus on?
* 


## Checklist
